app/Model/AppModel.php

```diff
@@ -849,7 +849,184 @@ public function delete($id = null, $cascade = true) {
     
     return $ret;
   }
-  
+
+  /**
+   * Duplicate objects.
+   *
+   * @since  COmanage Registry v3.2.0
+   * @param  Model   $model      Model to duplicate
+   * @param  String  $foreignKey The name of the foreign key from $model to its parent (or 'id' if no parent)
+   * @param  Integer $fkid       The ID of the foreign key in $model
+   * @param  Array   $idmap      Reference to an array of old IDs to new IDs (originals to duplicates)
+   * @param  Boolean $isTree     True if $model implements TreeBehavior
+   * @param  Boolean $alterName  if true we will change the name and description of the copied model.
+   * @return Boolean True on success
+   * @throws InvalidArgumentException
+   * @throws RuntimeException
+   */
+
```

Comment:
I'm not a huge fan of adding even more stuff to AppModel, but I suppose in v4 that's our least bad option. (In PE we'd probably want to use a trait instead.)

Reply:
With this change we keep the duplicate Object core functionality in one place.